spec/services/stats/r103_sar_business_unit_performace_report_spec.rb

```diff
@@ -115,38 +123,214 @@ module Stats
+                BGCD,"","",#{@bizgrp_cd.team_lead},"","",100.0,2,1,0,1,0,0.0,0,0,0,0,0,100.0,2,1,0,1,0,0.0,2,0,0,2,0
+                BGCD,DRCD,"",#{@dir_cd.team_lead},"","",100.0,2,1,0,1,0,0.0,0,0,0,0,0,100.0,2,1,0,1,0,0.0,2,0,0,2,0
+                BGCD,DRCD,RTC,#{@team_c.team_lead},"","",100.0,1,1,0,0,0,0.0,0,0,0,0,0,100.0,1,1,0,0,0,0.0,1,0,0,1,0
+                BGCD,DRCD,RTD,#{@team_d.team_lead},"","",0.0,1,0,0,1,0,0.0,0,0,0,0,0,0.0,1,0,0,1,0,0.0,1,0,0,1,0
+                Total,"","","","","",50.0,9,3,0,3,3,0.0,0,0,0,0,0,50.0,9,3,0,3,3,0.0,9,0,0,9,0
+              EOCSV
+              report = R103SarBusinessUnitPerformanceReport.new(period_start: Date.new(2017, 6, 5), period_end: Date.today, generate_bu_columns: true)
+              report.run
+              actual_lines = report.to_csv.map { |row| row.map(&:value) }
+              expected_lines = expected_text.split("\n")
+              actual_lines.zip(expected_lines).each do |actual, expected|
+                expect(CSV.generate_line(actual).chomp).to eq(expected)
+              end
             end
           end
         end
+  
+  
+        context 'with a case in the db that is unassigned' do
+          before do
+            Timecop.freeze Time.new(2017, 6, 30, 12, 0, 0) do
+              create :case, identifier: 'unassigned case'
+            end
+          end
+  
+          it 'does not raise an error' do
+            report = R103SarBusinessUnitPerformanceReport.new
+            expect { report.run }.not_to raise_error
+          end
+        end
+      end
+    end
 
-
-        if type.present?
-          refusal_reason = CaseClosure::RefusalReason.__send__(type)
-          kase.refusal_reason = refusal_reason
+    #rubocop:disable Metrics/ParameterLists
+    def create_case(received:, responded:, deadline:, team:, responder:, ident:, flagged: false, type: nil)
+      received_date = Date.parse(received)
+      responded_date = responded.nil? ? nil : Date.parse(responded)
+      kase = nil
+      Timecop.freeze(received_date + 10.hours) do
+        factory = :accepted_sar
+        kase = create factory, responding_team: team, responder: responder, identifier: ident, received_date: received_date
+        kase.external_deadline = Date.parse(deadline)
+        if flagged == true
+          CaseFlagForClearanceService.new(user: kase.managing_team.users.first, kase: kase, team: @team_dacu_disclosure).call
+        end
+        unless responded_date.nil?
+          Timecop.freeze responded_date + 14.hours do
+            kase.date_responded = Time.now
+            kase.state_machine.close!(acting_user: responder, acting_team: team)
+          end
         end
-        kase.save!
-        kase
       end
-      #rubocop:enable Metrics/ParameterLists
 
 
+      if type.present?
+        refusal_reason = CaseClosure::RefusalReason.__send__(type)
+        kase.refusal_reason = refusal_reason
+      end
+      kase.save!
```

Comment:
I think `kase.save!` will return the case anyway. Not sure we need a subsequent case.